tests/TestCase/ORM/EntityTest.php

```diff
@@ -46,6 +46,31 @@ public function testSetOneParamNoSetters()
         $this->assertSame(1, $entity->id);
         $this->assertEquals(1, $entity->getOriginal('id'));
         $this->assertEquals('bar', $entity->getOriginal('foo'));
+
+        $entity->set('level1.level2.level3', 'three');
+        $this->assertSame('three', $entity->level1->level2->level3);
+        $this->assertEquals('three', $entity->getOriginal('level1.level2.level3'));
+        $entity->set('level1.level2.level3', '333');
+        $this->assertEquals('three', $entity->getOriginal('level1.level2.level3'));
```

Comment:
It's worth noting that like "original" values, dirty state can only descend down to the last available entity.

```php
$e = new Entity(
    ['a' => ['b' => 'fun']], 
    ['markClean' => true]
);
var_dump($e->dirty('a')); // false

$e->set('a.b', 'times'); // Make an array element dirty.
var_dump($e->dirty('a')); // Definitely true still.

// True (because we stop at 'a' being dirty)? 
// Or Exception (because we can't descend into 'b' to _check_ for dirty)??
var_dump($e->dirty('a.b')); // ???
```